src/lib/utils/settings/abstract_setting.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+namespace opossum {
+
+/**
+ * This is an abstract class for all settings objects.
+ * A setting is a configuration knob of any component with a variable value.
+ * That component (or "parent") instantiates and destroys a setting.
```

Comment:
Parent makes me think of class hierarchies. Maybe "component configured by the setting"?